compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,46 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// make sure that we are in the condition to suggest the blanket implementation, if we are in the
+    /// case of suggest it, use the function `err_reporter` to report the error or suggestion.
+    // FIXME: The copy of TyCtxt it is a bad think in this case?
+    fn maybe_lint_blanket_trait_impl<T: rustc_errors::EmissionGuarantee>(
+        &self,
+        self_ty: &hir::Ty<'_>,
+        diagnostic: &mut DiagnosticBuilder<'_, T>,
+        _needs_bracket: bool,
+    ) {
+        let tcx = self.tcx();
+        let parent_id = tcx.hir().get_parent_item(self_ty.hir_id);
+        let child_ty = self_ty;
+        if let hir::Node::Item(hir::Item {
+            kind: hir::ItemKind::Impl(hir::Impl { self_ty: _, of_trait: Some(trait_ref), .. }),
+            ..
+        }) = tcx.hir().get_by_def_id(parent_id)
+        {
+            let trait_span = trait_ref.path.span;
+            let is_local = trait_ref.trait_def_id().map_or(false, |def_id| def_id.is_local());
+            if is_local {
+                let trait_name = tcx.sess.source_map().span_to_snippet(trait_span).unwrap();
+                let child_name = tcx.sess.source_map().span_to_snippet(child_ty.span).unwrap();
+                let blanket_msg = format!(
+                    "use a blanket implementation to implement {} for all types that also implement {:?}",
+                    trait_name, child_name
+                );
+                // FIXME: check support for the generics
+                let blanket_sugg = vec![
+                    (trait_span, "<T: ".to_owned()),
```

Comment:
you need to get the generics of the `hir::ItemKind::Impl` and then use [`hir::Generics::span_for_param_suggestion`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/struct.Generics.html#method.span_for_param_suggestion) for the right span to add `T: Trait`

Reply:
Thanks! done with the last version of the code, now I will debug the error that I have in the CI, and I will add a new test too specific for this test!